Tuprules.tup

```diff
@@ -12,23 +12,13 @@ else
 CC = gcc
 endif
 
-ifeq (@(TUP_DEBUG),y)
-CFLAGS += -g
-else
-CFLAGS += -Os
-endif
-
-ifdef AR
-AR = @(AR)
+CFLAGS += -D_FILE_OFFSET_BITS=64
+CFLAGS += -I$(TUP_CWD)/src
+ifdef CFLAGS
+CFLAGS += @(CFLAGS)
 else
-AR = ar
-endif
-
 CFLAGS += -W
```

Comment:
I think we should put the CFLAGS -W through -fno-common into the group with -I$(TUP_CWD)/src, so that they are always included. IMO these are important warnings that should be included in all profiles.

Since these flags would be before the 'ifdef CFLAGS' line, it is still possible for a user to override certain flags with their tup.config. For example, they could set CONFIG_CFLAGS to -Wno-switch-enum if they wanted to override the default -Wswitch-enum flag.

The 'else' block of the 'ifdef CFLAGS' can just be -O2 for the default optimization, since this is the most common thing that you are changing in the configs/*.config files.

This would also make gcc-dev.config and gcc-dev-lto.config much smaller.